datastore/src/main/java/com/ds/datastore/RequestListener.java

```diff
@@ -0,0 +1,24 @@
+package com.ds.datastore;
+
+import java.util.UUID;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.annotation.WebListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@WebListener
+public class RequestListener implements ServletRequestListener{
+    private Logger logger = LoggerFactory.getLogger(RequestListener.class);
+    @Override
+    public void requestInitialized(ServletRequestEvent sre){
+        ServletRequest request = sre.getServletRequest(); 
+        if(request.getAttribute("orderID") == null){
```

Comment:
The request attributes wont have the id - you need to get it from the header and then set it on the attributes. The request headers are sent by the client, and the request attributes are for setting attributes within the server. 

In order to get the headers, you will need to cast to `HttpServletRequest`

Im also wondering about the naming of `orderId` - the id might not be tied to an order. Normally this kind of id is called `requestId` or something similar.